### ISSUE TITLE
Wascally does not publish message after starting, stopping, and starting server during tests

### ISSUE THREAD

**Issue by krishnaglick** [LeanKit-Labs/wascally#123]
I'm using ActionHero.js and to make tests you start and stop the server each mocha describe block.
http://www.actionherojs.com/docs/#example-test

I have an initializer (just something that runs on server start) that sets up the rabbitMQ/Wascally stuff.

api.mq = require('wascally');
api.mq.configure(api.config.mq) (api.config.mq can be seen here: https://gist.github.com/krishnaglick/f45b79bc8cb745248441)

Within my test I setup a handler, publish a message, and confirm that the message was received. It works for test one but if I have an identical test called test two that gets run second then the message never gets published.

I was wondering if this is an issue with RabbitMQ, Wascally, or potentially ActionHero.

**Comment by derickbailey** [comment]
@krishnaglick do you have a test teardown method? if not, you probably need to add one, where you stop your existing wascally subscription before starting the next test. 

my best guess is that the wascally subscription is still alive, so the next test is sending a message but the handler from the previous test is picking up the message instead of the handler for the new test.

when you create your subscription, hold a reference to the result:

```js
var sub = wascally.handle(...);
```

and when you're done, release it:

```js
sub.remove();
```

do that in your test teardown scenario, if you're not already. hopefully that helps.

**Comment by krishnaglick** [comment]
I do have a handler cleanup method. All handlers in the app get added to a single object and in my afterEach block I call cleanupHandlers which iterates through each one and calls its remove method. I decided to double check to ensure that it was getting called correctly and it is.

I've also tried wascally.closeAll(true) and wascally.closeAll(false) in my after block to no effect. :(

**Comment by derickbailey** [comment]
can you post a gist with a complete test that reproduces the problem?

**Comment by krishnaglick** [comment]
In the process of whipping up a test project to give to you I discovered the error.
The way ActionHero initializes work is by running until next() is called. Looking through the code I discovered that the initializer for Wascally was calling next, however since Wascally's configure function is async it was actually getting called before the queue was properly configured!
Additionally it didn't like me called closeAll().

I removed my closeAll calls and fixed up the initializer so next() is called appropriately, and it works!

Thanks for the assistance!
